src/components/AutoUpdateTime.tsx

```diff
@@ -2,42 +2,35 @@
  * Displays the user's local time and updates it every minute.
  * The time auto-update logic is extracted to this component to avoid re-rendering a more complex component, e.g. DetailsPage.
  */
-import PropTypes from 'prop-types';
 import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
 import {View} from 'react-native';
 import DateUtils from '@libs/DateUtils';
 import useThemeStyles from '@styles/useThemeStyles';
+import {Timezone} from '@src/types/onyx/PersonalDetails';
 import Text from './Text';
-import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import withLocalize, {WithLocalizeProps} from './withLocalize';
 
-const propTypes = {
+type AutoUpdateTimeProps = WithLocalizeProps & {
     /** Timezone of the user from their personal details */
-    timezone: PropTypes.shape({
-        /** Value of selected timezone */
-        selected: PropTypes.string,
-
-        /** Whether timezone is automatically set */
-        automatic: PropTypes.bool,
-    }).isRequired,
-    ...withLocalizePropTypes,
+    timezone: Timezone;
 };
 
-function AutoUpdateTime(props) {
+function AutoUpdateTime({timezone, preferredLocale, translate}: AutoUpdateTimeProps) {
     const styles = useThemeStyles();
-    /**
-     * @returns {Date} Returns the locale Date object
-     */
-    const getCurrentUserLocalTime = useCallback(
-        () => DateUtils.getLocalDateFromDatetime(props.preferredLocale, null, props.timezone.selected),
-        [props.preferredLocale, props.timezone.selected],
-    );
+    /** @returns Returns the locale Date object */
+    const getCurrentUserLocalTime = useCallback(() => DateUtils.getLocalDateFromDatetime(preferredLocale, undefined, timezone.selected), [preferredLocale, timezone.selected]);
 
     const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime);
     const minuteRef = useRef(new Date().getMinutes());
-    const timezoneName = useMemo(() => DateUtils.getZoneAbbreviation(currentUserLocalTime, props.timezone.selected), [currentUserLocalTime, props.timezone.selected]);
+    const timezoneName = useMemo(() => {
+        if (timezone.selected) {
+            return DateUtils.getZoneAbbreviation(currentUserLocalTime, timezone.selected);
+        }
+        return '';
```

Comment:
Was this change inspired by a typing problem?

Reply:
Yes, as `getZoneAbbreviation` expects that timezone is defined